File: TP-Instrumentos-Frontend-main/src/types/Instrumento.ts
// Instrumento.ts
import { Categoria } from './Categoria';
export interface Instrumento {
    id: number
    instrumento: string
    marca: string
    modelo: string
    imagen: string
    precio: number
    costoEnvio: string
    cantidadVendida: number
    descripcion: string
    categoria: Categoria
}

File: TP-Instrumentos-Backend-main/src/main/java/utn/TpInstrumentosBackend/entities/Instrumento.java
package utn.TpInstrumentosBackend.entities;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@Entity
@SuperBuilder
@Table(name = "instrumento")
public class Instrumento extends Base{
    private String instrumento;
    private String marca;
    private String modelo;
    private String imagen;
    private double precio;
    private String costoEnvio;
    private int cantidadVendida;
    private String descripcion;

    @ManyToOne(optional = false)
    @JoinColumn(name = "categoria_id")
    private Categoria categoria;
}